packages/backend/src/open_payments/quote/service.ts

```diff
@@ -61,21 +61,30 @@ async function getQuote(
   return Quote.query(deps.knex).get(options).withGraphFetched('asset')
 }
 
-export interface CreateQuoteOptions {
+interface QuoteOptionsBase {
   paymentPointerId: string
-  sendAmount?: Amount
-  receiveAmount?: Amount
   receiver: string
   clientId?: string
 }
 
+interface QuoteOptionsWithSendAmount extends QuoteOptionsBase {
+  receiveAmount?: never
+  sendAmount?: Amount
+}
+
+interface QuoteOptionsWithReceiveAmount extends QuoteOptionsBase {
+  receiveAmount?: Amount
+  sendAmount?: never
+}
+
+export type CreateQuoteOptions =
+  | QuoteOptionsWithSendAmount
+  | QuoteOptionsWithReceiveAmount
+
 async function createQuote(
   deps: ServiceDependencies,
   options: CreateQuoteOptions
 ): Promise<Quote | QuoteError> {
-  if (options.sendAmount && options.receiveAmount) {
-    return QuoteError.InvalidAmount
-  }
```

Comment:
@raducristianpopa @wilsonianb 
I'm leaning on keeping this business logic sanity check here (even though the OpenAPI validator would prevent it). If there are some issues with validator we would still be ok in the service

Reply:
An alternative could be to move the check to `routes.ts` (like we had to do with the graphql resolver), but it'd make more sense to just keep the one check here.
So I vote we drop the resolver check in favor of this, and maybe we revisit if/when we have graphql `@oneOf`.

Reply:
:thinking: do we only change `CreateBody` then, and revert `CreateQuoteOptions`?
I think that'd avoid having to change the tests' `createOutgoingPayment`

Reply:
🤔 i think we should keep the changes for the CreateQuoteOptions type for consistency.

Reply:
My only reservation about that is we'd also need to check in the resolver.

Reply:
So we will need to keep
```ts
 if (args.input.sendAmount && args.input.receiveAmount) {
      return {
        code: errorToCode[QuoteError.InvalidAmount].toString(),
        success: false,
        message: errorToMessage[QuoteError.InvalidAmount]
      }
 }
```
in the resolver even though we check for `sendAmount` and `receiveAmount` in the service ?

Reply:
right, if we keep the changes to `CreateQuoteOptions`.

Reply:
I think the resolver will need the check in order to be able to call `quoteService.create` if the `options` type is:
```
export type CreateQuoteOptions =
  | QuoteOptionsWithSendAmount
  | QuoteOptionsWithReceiveAmount
```

Reply:
With the actual changes to the resolver:
https://github.com/interledger/rafiki/blob/1b5c01a4a984400406a81f3498bb5ed7ccaed452/packages/backend/src/graphql/resolvers/quote.ts#L33-L46
i was able to make a GraphQL request with both `sendAmount` and `receiveAmount`.
`quoteService.create` gets called properly even with the `options` type being
```ts
export type CreateQuoteOptions =
  | QuoteOptionsWithSendAmount
  | QuoteOptionsWithReceiveAmount
  ```

<img width="1074" alt="image" src="https://user-images.githubusercontent.com/60383558/211490921-0837c4ff-7628-4eeb-9d37-1a6a98c62e09.png">

Reply:
I've just enabled strict type checking for the `backend` package, but TypeScript it's still not complaining about it. 🤔

Reply:
I think it's not complaining because both `receiveAmount` and `sendAmount` are optional in their respective types